src/framework/FormatChecker.ts

```diff
@@ -50,46 +47,42 @@ export class FormatChecker extends Base {
     });
 
     if (this.checker instanceof RegExp) {
-      if (this.checker.test(message.cleanContent) === true) return;
-    } else if (this.checker instanceof Function) {
-      if (this.checker(message.cleanContent, logger)) return;
-    }
+      if (this.checker.test(message.cleanContent)) return;
+    } else if (this.checker(message, logger)) return;
 
     logger.debug('bad format detected');
 
     const { cleanContent, author } = message;
 
     await message.delete();
+    const plural = this.examples !== undefined && this.examples.length > 1;
+    const pluralSuffix = plural ? 's' : '';
```

Comment:
```ts
const plural = this.examples?.length > 1
const pluralSuffix = plural ? 's' : '';
```

Reply:
J'avais eu une erreur TypeScript, donc j'ai fais comme ça.
`TS2532: Object is possibly 'undefined'.`